x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/model_select_logic.test.ts

```diff
@@ -0,0 +1,133 @@
+  deploymentState: MlModelDeploymentState.NotDeployed,
+};
+const FETCH_MODELS_API_DATA_RESPONSE: MlModel[] = [
+  {
+    modelId: 'model_1',
+    title: 'Model 1',
+    type: 'ner',
+    deploymentState: MlModelDeploymentState.NotDeployed,
+    startTime: 0,
+    targetAllocationCount: 0,
+    nodeAllocationCount: 0,
+    threadsPerAllocation: 0,
+    isPlaceholder: false,
+    hasStats: false,
+  },
+];
+
+describe('ModelSelectLogic', () => {
+  const { mount } = new LogicMounter(ModelSelectLogic);
+  const { mount: mountCreateModelApiLogic } = new LogicMounter(CreateModelApiLogic);
+  const { mount: mountCachedFetchModelsApiLogic } = new LogicMounter(CachedFetchModelsApiLogic);
+  const { mount: mountStartModelApiLogic } = new LogicMounter(StartModelApiLogic);
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mountCreateModelApiLogic();
+    mountCachedFetchModelsApiLogic();
+    mountStartModelApiLogic();
+    mount();
+  });
+
+  describe('listeners', () => {
+    describe('createModel', () => {
+      it('creates the model', () => {
+        const modelId = 'model_1';
+        jest.spyOn(ModelSelectLogic.actions, 'createModelMakeRequest');
+
+        ModelSelectLogic.actions.createModel(modelId);
+
+        expect(ModelSelectLogic.actions.createModelMakeRequest).toHaveBeenCalledWith({ modelId });
+      });
+    });
+
+    describe('createModelSuccess', () => {
+      it('starts polling models', () => {
+        jest.spyOn(ModelSelectLogic.actions, 'startPollingModels');
+
+        ModelSelectLogic.actions.createModelSuccess(CREATE_MODEL_API_RESPONSE);
+
+        expect(ModelSelectLogic.actions.startPollingModels).toHaveBeenCalled();
+      });
+    });
+
+    describe('fetchModels', () => {
+      it('makes fetch models request', () => {
+        jest.spyOn(ModelSelectLogic.actions, 'fetchModelsMakeRequest');
+
+        ModelSelectLogic.actions.fetchModels();
+
+        expect(ModelSelectLogic.actions.fetchModelsMakeRequest).toHaveBeenCalled();
+      });
+    });
+
+    describe('startModel', () => {
+      it('makes start model request', () => {
+        const modelId = 'model_1';
+        jest.spyOn(ModelSelectLogic.actions, 'startModelMakeRequest');
+
+        ModelSelectLogic.actions.startModel(modelId);
+
+        expect(ModelSelectLogic.actions.startModelMakeRequest).toHaveBeenCalledWith({ modelId });
+      });
+    });
+
+    describe('startModelSuccess', () => {
+      it('starts polling models', () => {
+        jest.spyOn(ModelSelectLogic.actions, 'startPollingModels');
+
+        ModelSelectLogic.actions.startModelSuccess(CREATE_MODEL_API_RESPONSE);
+
+        expect(ModelSelectLogic.actions.startPollingModels).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('selectors', () => {
+    describe('areActionButtonsDisabled', () => {
+      it('is set to false if create and start APIs are idle', () => {
+        CreateModelApiLogic.actions.apiReset();
+        StartModelApiLogic.actions.apiReset();
```

Comment:
Is there a way to put the API in `Status.LOADING` state programmatically? I'd like to test that use case too.

Reply:
Ok, added a test.